### ISSUE TITLE
[BUG] Incompatible with the script manager "Userscripts" on Safari

### ISSUE THREAD

**Issue by OcileCollection** [tobyxdd/bread#2]
macOS Monterey Version 12.4
Safari Version 15.5
Userscripts Version 4.1.0

Issue: Not sure if the script is functioning. The code posted on OpenUserJS is the same as what was installed. Other scripts seem to be functional.

Is there some form of initiation I was missing?

**Comment by tobyxdd** [comment]
Is it not working on any websites at all or just some specific ones?

**Comment by tobyxdd** [comment]
I never tried it on Safari. Will get back to you after I do.

**Comment by tobyxdd** [comment]
Can you try it again with version 1.0.2? I guess some script managers don't support `match <all_urls>`

**Comment by tobyxdd** [comment]
Turns out Userscripts does not support `GM_registerMenuCommand`. I added an alternative keyboard shortcut "Ctrl+B" for opening the configuration dialog.

Can you try again with 1.0.4?

**Comment by tobyxdd** [comment]
3c1d4edee3f6c57375e4400a539633ca27f3faa7

**Comment by moritzschaefer** [comment]
@tobyxdd thanks. Ctrl+b works, yet the following piece of code leads to an error:
```

        acceptNode: (node) => {
            return (
                node.parentNode.nodeName !== 'SCRIPT' &&
                node.parentNode.nodeName !== 'NOSCRIPT' &&
                node.parentNode.nodeName !== 'STYLE' &&
                node.nodeValue.length >= minTextLength) ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_REJECT;
        }
```

Error: userscript:GM-Bread:1130 Uncaught TypeError: Cannot read property 'length' of null

Sugested fix could be to check for node.nodeValue !== null

**Comment by OcileCollection** [comment]
> macOS Monterey Version 12.4 Safari Version 15.5 Userscripts Version 4.1.0
> 
> Issue: Not sure if the script is functioning. The code posted on OpenUserJS is the same as what was installed. Other scripts seem to be functional.
> 
> Is there some form of initiation I was missing?

Issue has been resolved with Userscript Version 4.2.0 (52).
